qemu/tests/create_external_snapshot_via_protocol.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os
+import re
+import subprocess
+
+from qemu.tests import qemu_disk_img
+
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import error_context
+
+
+def run(test, params, env):
+
+    """
+    creat external snapshot using external drivers for backing file:
+    1). Prepare environment of ssh/https
+    2). Boot up vm and copy a file to it, then calculate its md5 value
+    3). Create external snapshot using ssh/https driver
+    4). Boot up vm using the external snapshot
+    5). Calculate the file's md5 value again and then compare
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def clean_key():
+
+        """
+        clear the ssh key
+        :return: None
+        """
+        process.system("rm -rf {}/id*".format(
+            params["ssh_key_dir"]), shell=True)
+
+    @error_context.context_aware
+    def setup_env_for_ssh():
+
+        """
+        Prepare ssh environment
+        :return: None
+        """
+        error_context.context("STEP 1: Prepare ssh login environment.")
+        clean_key()
```

Comment:
when run in a new host ,there is no key in the dir, to remove it so that if rerun it again,the dir is empty too.